src/hotspot/share/opto/loopUnswitch.cpp

```diff
@@ -24,38 +24,54 @@
 
 #include "precompiled.hpp"
 #include "memory/allocation.inline.hpp"
+#include "opto/castnode.hpp"
+#include "opto/cfgnode.hpp"
 #include "opto/connode.hpp"
 #include "opto/convertnode.hpp"
 #include "opto/loopnode.hpp"
 #include "opto/opaquenode.hpp"
 #include "opto/predicates.hpp"
 #include "opto/rootnode.hpp"
 
-//================= Loop Unswitching =====================
+// Loop Unswitching is a loop optimization to move an invariant, non-loop-exiting test in the loop body before the loop.
+// Such a test is either always true or always false in all loop iterations and could therefore only be executed once.
+// To achieve that, we duplicate the loop and change the original and cloned loop as follows:
+// - Original loop -> true-path-loop: The true-path of the invariant, non-loop-exiting test in the original loop
+//                                    is kept while the false-path is killed. We call this unswitched loop version
+//                                    the true-path-loop.
+// - Cloned loop -> False-path-loop:  The false-path of the invariant, non-loop-exiting test in the cloned loop
+//                                    is kept while the true-path is killed. We call this unswitched loop version
+//                                    the false-path loop.
```

Comment:
Comments are too wide and difficult to read. Suggestion is to put explanation on new line with shift:
```
// - Original loop -> true-path-loop:
//        The true-path of the invariant, non-loop-exiting test in the original loop
//        is kept while the false-path is killed. We call this unswitched loop version
//        the true-path-loop.
```